protocol/src/main/java/org/mifos/chatbot/protocol/facebookMessenger/FacebookMessengerChatService.java

```diff
@@ -140,80 +144,101 @@ private static String base64Encode(String input) {
-                                senderId
-                        );
-                    }
-                    sendTextMessage(senderId, "Login successfully.");
-                } else {
-                    sendTextMessage(senderId, "Please enter valid credentials.");
-                    return;
+                sendTextMessage(senderId, "Sorry i didn't get that.");
+            }
+        } else {
+            sendTextMessage(senderId, "Your previous credentials are not correct. Please login again.");
+            userRepository.removeUser(username);
+        }
+    }
+
+    private void handleLogin(String senderId, String messageText) {
+        StringBuilder username = new StringBuilder();
+        StringBuilder password = new StringBuilder();
+        String creds = messageText.replaceAll("login:", "");
+        //extracting username and password from message
+        for (int i = 0; i < creds.length(); i++) {
+            if (creds.charAt(i) == ':') {
+                for (i++; i < creds.length(); i++) {
+                    password.append(creds.charAt(i));
                 }
+                break;
             }
-            return;
+            username.append(creds.charAt(i));
         }
-        User user = userRepository.findUserByFBID(senderId);
-        if(user != null) {
-            String username = user.getUsername();
-            String password = user.getSecret_Pass();
-            if(authUser(username, password)) {
-                ApiClient apiClient = new ApiClient(base64Encode(username + ":" + password));
-                org.mifos.chatbot.client.Configuration.setDefaultApiClient(apiClient);
-                List<MifosResponse> responseList = adapterService.handle(messageText.toLowerCase());
-                if (!responseList.isEmpty()) {
-                    for (MifosResponse response : responseList) {
-                        sendTextMessage(senderId, response.getContent());
-                    }
+        if(userRepository.FBIDExist(senderId)) {
+            sendTextMessage(senderId, "You are already logged in, please logout first.");
+        } else {
+            if (authUser(username.toString(), password.toString())) {
+                User user = userRepository.findUserByUsername(username.toString());
+                if (user == null) {
+                    userRepository.addUserByFBID(
+                            username.toString(),
+                            base64Encode(password.toString()),
+                            senderId
+                    );
                 } else {
-                    sendTextMessage(senderId, "Sorry i didn't get that.");
+                    userRepository.updateUserFBID(
+                            username.toString(),
+                            base64Encode(password.toString()),
+                            senderId
+                    );
                 }
+                sendTextMessage(senderId, "Login successfully.");
```

Comment:
Login successful or Logged in successfully, just minor grammatical nitpicking

Reply:
done.